[WIP] test: add getaddress mock

dougsland opened the pull request:
There are test pipelines environments that cannot use external network due security restrictions. This patch ensure the code can still be tested.

Fixes: https://github.com/containers/hirte/issues/398

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>
Signed-off-by: Michael Engel <mengel@redhat.com>

dougsland committed:
test: add getaddress mock

There are test pipelines environments that cannot use external
network due security restrictions. This patch ensure the code
can be still tested.

Fixes: https://github.com/containers/hirte/issues/398

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>
Signed-off-by: Michael Engel <mengel@redhat.com>

mwperina commented on src/libhirte/test/common/network/get_address_test.c:
```diff
@@ -22,13 +57,15 @@ bool test_get_address_failure(const char *domain) {
                 return false;
         }
+
+
```
What are those 2 empty lines for?

dougsland replied:
nothing, it was late in the night ;-)

mwperina commented on src/libhirte/test/common/network/get_address_test.c:
```diff
@@ -7,10 +7,45 @@
+        return -1;
+}
+
+int getaddrinfo_func_mock(const char *, const char *, const struct addrinfo *, struct addrinfo **foo) {
```
Just wondering if it wouldn't be better to have more complete mock definition?

1. Have a mocking parameters and results structure:
````
struct getaddrinfo_mock_params {
    char* hostname;
    char** resolved_addresses;
    int result;
}
````
2. Initialize this structure for all test cases:
````
static struct getaddrinfo_mock_params[] test_data = [
    { NULL, NULL, -1 },
    { ".", NULL, -1 },
    { "redhat", NULL, -1 },
    { "?10.10.10.3", NULL -1 },
    { "192.168.1.", NULL, -1 },
    { "8.8.8.8", ["8.8.8.8"], 0},
    { "localhost.localdomain", ["127.0.01", "::1"], 0}
];
````
3. Having only `getaddrinfo_func_mock`, which uses `test_data` array and create output according to the input and `test_data` content

sandrobonazzola committed:
Add coverage report generation for tests

Added targets to Makefile for getting a coverage report from the unit
tests.

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>

sandrobonazzola committed:
update after review for PR 436

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>

sandrobonazzola committed:
build-base: add tools for coverage reports

- adding lcov as gcov is not available on el9
- adding html2text to generate summary for github action step

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>

sandrobonazzola committed:
containers minimization

- avoid to install docs
- cleanup dnf cache
- avoid generating unnecessary intermediate layers

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>

mwperina committed:
Use correct architecture within container builds (#439)

Use correct architecture when building containers on non-x86_64
platforms.

Fixes: https://github.com/containers/hirte/issues/437

Signed-off-by: Martin Perina <mperina@redhat.com>

sandrobonazzola committed:
follow up on PR 440 review comments

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>

mwperina committed:
Fix detecting architecture within container builds

We need to detect the architecture to provide the correct DNF repository
with BlueChi RPMs.

Fixes: https://github.com/containers/hirte/issues/437
Signed-off-by: Martin Perina <mperina@redhat.com>

sandrobonazzola committed:
unit-tests: generate coverage for unit tests

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>
